rustls/src/msgs/handshake.rs

```diff
@@ -890,12 +890,20 @@ impl Codec for ClientHelloPayload {
 }
 
 impl ClientHelloPayload {
+    pub fn get_extensions(&self) -> &[ClientExtension] {
```

Comment:
Notice how the return value encodes "no extensions" as an empty slice instead of as an `Option`.